haystack/document_stores/weaviate.py

```diff
@@ -401,6 +407,12 @@ def write_documents(
         for do in document_objects:
             do.id = self._sanitize_id(id=do.id, index=index)
 
+            # Convert pd.DataFrames to json string to be indexed to Weaviate
+            if do.content_type == "table":
+                # Deduplicate column names
+                do.content.columns = pd.io.parsers.base_parser.ParserBase({"usecols": None})._maybe_dedup_names(do.content.columns)
+                do.content = do.content.to_json()
```

Comment:
Why do we need the deduplication of column names here but not in other document stores? I would have expected such a function rather in the `Document` class itself than here as it should contain all the main logic.

Also wondering why we need the `to_json` here as we later (~ line 440) convert the doc to dict anyway :thinking:

Reply:
Weaviate only allows to index strings, so this `to_json` method here is converting the `pd.DataFrame` to a JSON-string in order to able to index the table. Before calling the DataFrame's `to_json` method, we need to make sure that we do not have duplicate column names, because otherwise we could have duplicate keys in the resulting JSON object. The `to_dict` method we call in line 440 is then converting the `Document` object (with a JSON-string inside the content field if content_type is "table") to a dictionary.

For the other DocumentStores, we are either indexing the plain DataFrame (`InMemoryDocumentStore`) or transforming the DataFrame into a list of lists of strings. Therefore, no deduplication is needed in these cases.

Reply:
I changed this such that we use the `Document`'s `to_dict` method to transform a pd.DataFrame into a nested list.